packages/axelar-std/tests/event.rs

```diff
@@ -0,0 +1,199 @@
+use soroban_sdk::testutils::Address as _;
+use soroban_sdk::{contract, contractimpl, Address, Env, String};
+
+mod into_event {
+    use stellar_axelar_std::events::{fmt_last_emitted_event, Event};
+    use stellar_axelar_std::IntoEvent;
+
+    use super::*;
+
+    #[contract]
+    pub struct Contract;
+
+    #[derive(Debug, PartialEq, Eq, IntoEvent)]
+    struct EmptyEvent;
+
+    #[derive(Debug, PartialEq, Eq, IntoEvent)]
+    struct SingleDataEvent {
+        topic1: Address,
+        topic2: Address,
+        #[data]
+        data: String,
+    }
+
+    #[derive(Debug, PartialEq, Eq, IntoEvent)]
+    #[event_name("custom_name")]
+    struct NamedEvent {
+        topic: Address,
+        #[data]
+        data: String,
+    }
+
+    #[derive(Debug, PartialEq, Eq, IntoEvent)]
+    struct MultiDataEvent {
+        topic1: String,
+        topic2: Address,
+        #[data]
+        data1: String,
+        #[data]
+        data2: String,
+    }
+
+    #[derive(Debug, PartialEq, Eq, IntoEvent)]
+    struct NoDataEvent {
+        topic1: String,
+        topic2: Address,
+    }
+
+    #[derive(Debug, PartialEq, Eq, IntoEvent)]
+    struct NoTopicEvent {
+        #[data]
+        data1: String,
+        #[data]
+        data2: Address,
+        #[data]
+        data3: String,
+    }
+
+    #[contractimpl]
+    impl Contract {
+        pub const fn __constructor() {}
+
+        pub fn empty(env: &Env) {
+            EmptyEvent.emit(env);
+        }
+
+        pub fn single_data(env: &Env, topic1: Address, topic2: Address, data: String) {
+            SingleDataEvent {
+                topic1,
+                topic2,
+                data,
+            }
+            .emit(env);
+        }
+
+        pub fn named(env: &Env, topic: Address, data: String) {
+            NamedEvent { topic, data }.emit(env);
+        }
+
+        pub fn multi_data(
+            env: &Env,
+            topic1: String,
+            topic2: Address,
+            data1: String,
+            data2: String,
+        ) {
+            MultiDataEvent {
+                topic1,
+                topic2,
+                data1,
+                data2,
+            }
+            .emit(env);
+        }
+
+        pub fn no_data(env: &Env, topic1: String, topic2: Address) {
+            NoDataEvent { topic1, topic2 }.emit(env);
+        }
+
+        pub fn no_topic(env: &Env, data1: String, data2: Address, data3: String) {
+            NoTopicEvent {
+                data1,
+                data2,
+                data3,
+            }
+            .emit(env);
+        }
+    }
+
+    #[test]
+    fn event_empty_succeeds() {
+        let env = &Env::default();
+
+        let contract_id = env.register(Contract, ());
+        let client = ContractClient::new(env, &contract_id);
+
+        client.empty();
+
+        goldie::assert!(fmt_last_emitted_event::<EmptyEvent>(env));
+    }
+
+    #[test]
+    fn event_single_data_succeeds() {
```

Comment:
```suggestion
    fn single_data_succeeds() {
```
Or single_data_event_emitted